### PR TITLE
Improve code blocks in testing guide

### PR DESCRIPTION
**Pull request by azinneera**
## Purpose
Improve code blocks in testing guide
## Check List

- [ ] **Page Addition**
  - [ ] Add `permalink` to pages
  - [ ] If contains empty folder(s), Add front-matter `redirect_to:`

- [ ] **Page Rename**
  - [ ] Add front-matter `redirect_from`
  - [ ] Add front-matter `redirect_to:` (If applicable)

### CODE REVIEW

File: swan-lake/learn/mocking.md [hunk 1]
```diff
@@ -83,112 +87,114 @@ import ballerina/stringutils;
...
-    test:assertEquals(result, "When Sheldon wants an egg, he cracks open a chicken.");
+    // invoke the function to test
+    string result = checkpanic getRandomJoke("Sanjiva");
+    // verify the return value against the expected string
+    test:assertEquals(result, "When Sanjiva wants an egg, he cracks open a chicken.");
 }
 
-// stubbing to return different values for each function call
+// This test stubs the behavior of the `get` function to return a specified sequence of values
+// for each `get` function invocation  when testing the `getRandomJoke` function
```
Comment (praneesha) [inline swan-lake/learn/mocking.md:159]:
```suggestion
// for each `get` function invocation when testing the `getRandomJoke` function
```

File: swan-lake/learn/mocking.md [hunk 2]
```diff
@@ -83,112 +87,114 @@ import ballerina/stringutils;
...
 public function testReturnSequence() {
-	clientEndpoint = <http:Client>test:mock(http:Client);
-	test:prepare(clientEndpoint).when("get").thenReturnSequence(getCategoriesResponse(), getMockResponse());
-
-	string result = checkpanic getRandomJoke("Sheldon");
-	test:assertEquals(result, "When Sheldon wants an egg, he cracks open a chicken.");
-
+    // create a default mock HTTP Client and assign it to the `clientEndpoint` object
+    clientEndpoint = <http:Client>test:mock(http:Client);
+    // stub to return the corresponding value for each invocation. i.e. the first call to `get` will return the mock
```
Comment (praneesha) [inline swan-lake/learn/mocking.md:164]:
```suggestion
    // stub to return the corresponding value for each invocation i.e., the first call to `get` will return the mock
```

File: swan-lake/learn/mocking.md [hunk 3]
```diff
@@ -83,112 +87,114 @@ import ballerina/stringutils;
...
+    test:assertEquals(result, "When Sanjiva wants an egg, he cracks open a chicken.");
 }
 
-// stubbing a member variable value
+// This function shows how to stub a member variable value of the HTTP Client object
 @test:Config {}
 function testMemberVariable() {
+    // create a default mock HTTP Client and assign it to the `clientEndpoint` object
     clientEndpoint = <http:Client>test:mock(http:Client);
+    // stub the value of `url` variable to return the specified string
```
Comment (praneesha) [inline swan-lake/learn/mocking.md:178]:
```suggestion
    // stub the value of the `url` variable to return the specified string
```

File: swan-lake/learn/mocking.md [hunk 4]
```diff
@@ -83,112 +87,114 @@ import ballerina/stringutils;
...
     test:prepare(clientEndpoint).getMember("url").thenReturn("https://foo.com/");
+    // verify the specified value is set
```
Comment (praneesha) [inline swan-lake/learn/mocking.md:180]:
```suggestion
    // verify if the specified value is set
```

File: swan-lake/learn/mocking.md [hunk 5]
```diff
@@ -83,112 +87,114 @@ import ballerina/stringutils;
...
     test:assertEquals(clientEndpoint.url, "https://foo.com/");
 }
 
-// util function for constructing a mock response
+// Returns a mock HTTP response to be used for random joke API invocation
```
Comment (praneesha) [inline swan-lake/learn/mocking.md:184]:
```suggestion
// Returns a mock HTTP response to be used for random joke API invocation
```
If we are starting with a capital letter please do the same in all comments for consistency.

File: swan-lake/learn/mocking.md [hunk 6]
```diff
@@ -83,112 +87,114 @@ import ballerina/stringutils;
...
-	mockResponse.setPayload(mockPayload);
-	return mockResponse;
+    http:Response mockResponse = new;
+    json mockPayload = {"value":"When Chuck Norris wants an egg, he cracks open a chicken."};
+    mockResponse.setPayload(mockPayload);
+    return mockResponse;
 }
 
-// util function for constructing a mock response
+// Returns a mock response to be used for category API invocation
```
Comment (praneesha) [inline swan-lake/learn/mocking.md:192]:
```suggestion
// Returns a mock response to be used for the category API invocation
```

File: swan-lake/learn/mocking.md [hunk 7]
```diff
@@ -197,40 +203,46 @@ function getCategoriesResponse() returns http:Response {
...
 }
 ```
 **main_test.bal**
+
 ```ballerina
 // main_test.bal
 import ballerina/test;
+import ballerina/email;
 
+// This function stubs the `send` function of the SmtpClient client object to do nothing when invoked
```
Comment (praneesha) [inline swan-lake/learn/mocking.md:235]:
```suggestion
// This function stubs the `send` function of the `SmtpClient` client object to do nothing when invoked
```

File: swan-lake/learn/mocking.md [hunk 8]
```diff
@@ -197,40 +203,46 @@ function getCategoriesResponse() returns http:Response {
...
+// for testing the `sendNotification` function
 @test:Config {}
 function testSendNotification() {
-	smtpClient = <email:SmtpClient>test:mock(email:SmtpClient);
-	test:prepare(smtpClient).when("send").doNothing();
-	string[] emailIds = ["user1@test.com", "user2@test.com"];
-
-	test:assertEquals(sendNotification(emailIds), ());
+    // create a default mock Smtp Client and assign it to the `smtpClient` object
```
Comment (praneesha) [inline swan-lake/learn/mocking.md:239]:
```suggestion
    // create a default mock SMTP client and assign it to the `smtpClient` object
```

File: swan-lake/learn/mocking.md [hunk 9]
```diff
@@ -197,40 +203,46 @@ function getCategoriesResponse() returns http:Response {
...
+    smtpClient = <email:SmtpClient>test:mock(email:SmtpClient);
+    // stub to do nothing when `send` function is invoked
```
Comment (praneesha) [inline swan-lake/learn/mocking.md:241]:
```suggestion
    // stub to do nothing when the`send` function is invoked
```

File: swan-lake/learn/mocking.md [hunk 10]
```diff
@@ -252,34 +264,43 @@ Mocking an imported function will apply the mocked function to every instance of
...
+
 **main_test.bal**
+
 ```ballerina
 // main.test.bal
 import ballerina/test;
 
 (any|error)[] outputs = [];
 
+// This is the mock function that replaces the `io:println` function
```
Comment (praneesha) [inline swan-lake/learn/mocking.md:287]:
```suggestion
// This is the mock function, which replaces the `io:println` function
```

File: swan-lake/learn/mocking.md [hunk 11]
```diff
@@ -252,34 +264,43 @@ Mocking an imported function will apply the mocked function to every instance of
...
 @test:Mock {
-	moduleName: "ballerina/io",
-	functionName: "println"
+    moduleName: "ballerina/io",
+    functionName: "println"
 }
 function mockIoPrintLn((any|error)... text) {
-	outputs.push(text);
+    // Append print statement to a global array
```
Comment (praneesha) [inline swan-lake/learn/mocking.md:293]:
```suggestion
    // Append the print statement to a global array
```

File: swan-lake/learn/mocking.md [hunk 12]
```diff
@@ -291,44 +312,59 @@ The object specified with the `@test:MockFn {}` annotation will be considered as
...
 @test:Config {}
 function testCall() {
-	test:when(intAddMockFn).call("mockIntAdd");
-	test:assertEquals(addValues(11, 6), 5, msg = "function mocking failed");
+    // stub to call another function when `intAdd` is called
+    test:when(intAddMockFn).call("mockIntAdd");
+    test:assertEquals(addValues(11, 6), 5, msg = "function mocking failed");
 }
 
+// Mock function to be used in place of `intAdd` function
```
Comment (praneesha) [inline swan-lake/learn/mocking.md:365]:
```suggestion
// Mock function to be used in place of the `intAdd` function
```

File: swan-lake/learn/testing-quick-start.md [hunk 13]
```diff
@@ -11,14 +11,11 @@ redirect_from:
...
 ---
 
-Ballerina Language has a built-in robust test framework, which allows you to achieve multiple levels of the test pyramid including unit testing, integration testing, and end to end testing.  It provides features such as assertions, data providers, mocking, and code coverage, which enables the programmers to write comprehensive tests.
-
-
 # Quick Start
 
-Let’s write a simple Ballerina function to test it.
-
+Ballerina Language has a built-in robust test framework, which allows you to achieve multiple levels of the test pyramid including unit testing, integration testing, and end to end testing.  It provides features such as assertions, data providers, mocking, and code coverage, which enables the programmers to write comprehensive tests.
```
Comment (praneesha) [inline swan-lake/learn/testing-quick-start.md:16]:
```suggestion
The Ballerina Language has a built-in robust test framework, which allows you to achieve multiple levels of the test pyramid including unit testing, integration testing, and end to end testing.  It provides features such as assertions, data providers, mocking, and code coverage, which enable the programmers to write comprehensive tests.
```

File: swan-lake/learn/testing-quick-start.md [hunk 14]
```diff
@@ -50,22 +47,22 @@ For more information on the command, see [Structuring Ballerina Code](/swan-lake
     
     http:Client clientEndpoint = new("https://api.chucknorris.io/jokes/");
     
-    // This function performs a `get` request to the Chuck Norris API and
-    // returns a random joke with the name replaced with the provided name
+    // This function performs a `get` request to the Chuck Norris API and returns a random joke 
+    // with the name replaced with the provided name or an error if API invocation fails.
```
Comment (praneesha) [inline swan-lake/learn/testing-quick-start.md:51]:
```suggestion
    // with the name replaced with the provided name or an error if the API invocation fails.
```

File: swan-lake/learn/testing-quick-start.md [hunk 15]
```diff
@@ -79,29 +76,28 @@ For more information on the command, see [Structuring Ballerina Code](/swan-lake
     import ballerina/test;
     import ballerina/http;
     
+    // This test function tests the behavior of the `getRandomJoke` when
+    // the API returns a success response
     @test:Config {}
     function testGetRandomJoke() {
+        // create a default mock HTTP Client and assign it to the clientEndpoint
```
Comment (praneesha) [inline swan-lake/learn/testing-quick-start.md:83]:
```suggestion
        // create a default mock HTTP Client and assign it to the `clientEndpoint`
```

File: swan-lake/learn/testing-quick-start.md [hunk 16]
```diff
@@ -79,29 +76,28 @@ For more information on the command, see [Structuring Ballerina Code](/swan-lake
...
-        string result = checkpanic getRandomJoke("Sheldon");
+        // invoke the function to test
+        string result = checkpanic getRandomJoke("Sanjiva");
         io:println(result);
-        test:assertEquals(result, "When Sheldon wants an egg, he cracks open a chicken.");
+        // verify the return value   
+        test:assertEquals(result, "When Sanjiva wants an egg, he cracks open a chicken.");
     }
     
+    // Return a mock HTTP response for random joke API invocation
```
Comment (praneesha) [inline swan-lake/learn/testing-quick-start.md:94]:
```suggestion
    // Return a mock HTTP response for the `random joke` API invocation
```

File: swan-lake/learn/testing-quick-start.md [hunk 17]
```diff
@@ -119,7 +115,7 @@ For more information on the command, see [Structuring Ballerina Code](/swan-lake
     
     Running Tests with Coverage
         foo/joke:0.1.0
-    When Sheldon wants an egg, he cracks open a chicken.
+    When Sanjiva wants an egg, he cracks open a chicken.
```
Comment (praneesha) [inline swan-lake/learn/testing-quick-start.md:118]:
```suggestion
    When Sanjiva wants an egg, he cracks open a chicken.
```
Let's reconsider this.

File: swan-lake/learn/mocking.md (hunk 5 above)
Comment (azinneera) [reply swan-lake/learn/mocking.md:184]: Capital letters are used for only function descriptions here.

Comment (praneesha) [reply swan-lake/learn/mocking.md:184]: ok

File: swan-lake/learn/testing-quick-start.md (hunk 17 above)
Comment (azinneera) [reply swan-lake/learn/testing-quick-start.md]: Changed the name in 8e258b9

File: swan-lake/learn/testing-quick-start.md (hunk 16 above)
Comment (azinneera) [reply swan-lake/learn/testing-quick-start.md]: Fixed in 8e258b9

**Pull request merged**